### PR TITLE
More GitHub Actions improvements

### PR DESCRIPTION
**Pull request by swissspidy**
Trying to speed up things and avoid unnecessary runs...

### CODE REVIEW

File: .github/workflows/deploy-storybook.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ name: Deploy to GH Pages
 
 on:
   push:
+    # Don't run for docs-only changes.
+    paths-ignore:
+      - 'docs/**'
```
Comment (spacedmonkey) [inline .github/workflows/deploy-storybook.yml:7]:
```suggestion 
      - 'docs/**'
      - '.github/**'
      - '.wordpress-org/**'
```

File: .github/workflows/lint-php.yml [hunk 2]
```diff
@@ -2,10 +2,26 @@ name: Lint PHP
 
 on:
   push:
+    # Only run if PHP-related files changed.
+    paths:
+      - '**.php'
```
Comment (spacedmonkey) [inline .github/workflows/lint-php.yml:7]: Would this cover `scoper.inc.php`?

Comment (swissspidy) [reply .github/workflows/lint-php.yml:7]: [Yes.](https://www.digitalocean.com/community/tools/glob?comments=true&glob=%2A%2A.php&matches=false&tests=scoper.inc.php)

File: .github/workflows/tests-unit-php.yml [hunk 3]
```diff
@@ -2,10 +2,26 @@ name: PHP Unit Tests
...
 on:
   push:
+    # Only run if PHP-related files changed.
+    paths:
+      - '**.php'
+      - 'phpunit.xml.dist'
+      - 'phpunit-multisite.xml.dist'
+      - 'composer.json'
+      - 'composer.lock'
+      - 'tests/phpunit/**'
```
Comment (spacedmonkey) [inline .github/workflows/tests-unit-php.yml:12]:
```suggestion
      - 'tests/phpunit/**'
      - 'includes/**'
```

File: .github/workflows/tests-unit-php.yml [hunk 4]
```diff
@@ -2,10 +2,26 @@ name: PHP Unit Tests
...
       - release/*
   pull_request:
+    # Only run if PHP-related files changed.
+    paths:
+      - '**.php'
+      - 'phpunit.xml.dist'
+      - 'phpunit-multisite.xml.dist'
+      - 'composer.json'
+      - 'composer.lock'
+      - 'tests/phpunit/**'
```
Comment (spacedmonkey) [inline .github/workflows/tests-unit-php.yml:24]:
```suggestion
      - 'tests/phpunit/**',
      - 'includes/**'
```

File: .github/workflows/tests-unit-php.yml (hunk 3 above)
Comment (swissspidy) [reply .github/workflows/tests-unit-php.yml:12]: No commas.

`includes/` should be covered by `**.php` at the top.

File: .github/workflows/tests-unit-js.yml [hunk 5]
```diff
@@ -2,10 +2,24 @@ name: JavaScript Unit Tests
 
 on:
   push:
+    # Only run if JS-related files changed.
+    paths:
+      - '**.js'
+      - 'package.json'
+      - 'package-lock.json'
```
Comment (spacedmonkey) [inline .github/workflows/tests-unit-js.yml:9]: Do we need 
` - 'packages/**/package.json' `?

Comment (swissspidy) [reply .github/workflows/tests-unit-js.yml:9]: I'll just need to replace `package.json` with `**/package.json`. ([as per glob tool](https://www.digitalocean.com/community/tools/glob?comments=true&glob=%2A%2A%2Fpackage.json&matches=false&tests=packages%2Fi18n%2Fpackage.json&tests=package.json))

File: .github/workflows/tests-unit-php.yml (hunk 3 above)
Comment (spacedmonkey) [reply .github/workflows/tests-unit-php.yml:12]: What about https://github.com/google/web-stories-wp/blob/main/includes/data/stories/demo.json ?

File: .github/workflows/tests-unit-js.yml (hunk 5 above)
Comment (spacedmonkey) [reply .github/workflows/tests-unit-js.yml:9]: Yeah, just wonder.. I guess not.

**Comment by spacedmonkey** [comment]
I think we should also change the e2e tests to only run if 
```
 paths:
      - '**.js'
      - '**.cjs'
      - '**.css'
      - '**.php'
      - 'phpcs.xml.dist'
      - 'phpmd.xml'
      - 'phpstan.neon.dist'
      - 'composer.json'
      - 'composer.lock'
      - 'packages/**/*.md'
      - '.eslint*'
      - '.markdownlint*'
      - '.nvmrc'
      - '.prettier*'
      - '.stylelint*'
      - 'package.json'
      - 'package-lock.json'
 ```

File: .github/workflows/tests-unit-php.yml (hunk 3 above)
Comment (swissspidy) [reply .github/workflows/tests-unit-php.yml:12]: You're right. It's used by the `test_json_file_contains_replaceable_urls` test

**Pull request merged**